include/bluetooth/mesh/main.h

```diff
@@ -230,6 +273,38 @@ int bt_mesh_input_string(const char *str);
  */
 int bt_mesh_input_number(uint32_t num);
 
+/** @brief Provide Device public key.
+ *
+ *  This is intended to be called after the @ref bt_mesh_prov::oob_pub_key
+ *  callback has been called.
+ *
+ *  @param public_key Device public key.
+ *
+ *  @return Zero on success or (negative) error code otherwise.
+ */
+int bt_mesh_prov_set_public_key(const uint8_t public_key[64]);
```

Comment:
```suggestion
int bt_mesh_prov_remote_pub_key_set(const uint8_t public_key[64]);
```

Also, need to document that this is a provisioner only API.